src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Map<String, Integer> storage = new HashMap<>();
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+            handler.handle(transaction, storage);
+        }
+    }
+
+    @Override
+    public Map<String, Integer> getStorage() {
+        return storage;
+    }
```

Comment:
The getStorage() method currently violates the encapsulation principle by returning a reference to the storage map, which allows for its modification outside of the ShopServiceImpl class. Consider returning a copy of the storage map or an unmodifiable view of the map.